rules/creates-intercom-user.md

```diff
@@ -0,0 +1,53 @@
+---
+gallery: true
+short_description: On first login/signup call the Intercom API to record the contact as a new User
+categories:
+- webhook
+---
+## Creates a new user in Intercom on First Login/SignUp
+
+This rule will check if this is the first login of the user, and will call the Intercom API to save the contact as a new user. A persistent property named `signedUp` is used to track whether this is the first login or not.
+
+```
+function (user, context, callback) {
+  var moment = require('moment-timezone');
+  user.app_metadata = user.app_metadata || {};
+  
+  // Only create a new intercom user when user signs up
+  if (!user.app_metadata.signedUp) {
+    var data = {
+      "user_id": user.user_id,
+      "email": user.email,
+      "name": user.name,
+      "signed_up_at": moment(user.created_at).unix(),
+      "last_seen_ip" : context.request.ip,
+      "last_seen_user_agent": context.request.userAgent
+    };
+    var accessToken = 'YOUR INTERCOM ACCESS TOKEN';
+
+    request.post({
+      url: 'https://api.intercom.io/users',
+      headers: {
+        'Authorization': 'Bearer ' + accessToken,
+        'Accept': 'application/json'
+      },
+      json: data
+    }, function (err, res, body) {
+      if (err) return callback(err);
```

Comment:
This error handling will fail all user logins if the Intercom API is down, for example. Should we perhaps allow logins to continue despite API failures, and simply update the Intercom API based on the `app_metadata._intercom_signup` (or whatever name we choose) property alone?